scripts/staking_payouts.js

```diff
@@ -0,0 +1,367 @@
+    ).unwrapOrDefault();
+
+    return differenceWith((a, b) => a.eq(b), allEras, ledger.claimedRewards);
+  };
+
+  return from(validators).pipe(
+    mergeMap(
+      (stashAddress) =>
+        from(singleStashEras(api, stashAddress, eras)).pipe(
+          filterRx(complement(isEmpty)),
+          mapRx(assoc("eras", __, { stashAddress }))
+        ),
+      ConcurrentRequestsLimit
+    )
+  );
+};
+
+/**
+ * Signs and sends extrinsics produced by the given observable.
+ *
+ * @param {*} api
+ * @param {*} initiator
+ * @param {Rx.Observable<import("@polkadot/types/interfaces").Extrinsic>} payouts$
+ *
+ * @returns {Rx.Observable<string>} executed tx with hashes
+ */
+const signAndSendPayouts = curry((api, initiator, payouts$) => {
+  const nonces$ = timer(0, 1000).pipe(
+    switchMap(() =>
+      from(api.rpc.system.accountNextIndex(initiator.address)).pipe(
+        repeat(),
+        scan((nonce) => nonce.add(new BN(1)))
+      )
+    )
+  );
+
+  const sendTx = signAndSendExtrinsic(initiator, { finalize: true });
+
+  return zip(payouts$, nonces$).pipe(
+    mergeMap(
+      ([tx, nonce]) =>
+        from(
+          Promise.race([
+            sendTx({ nonce }, tx),
+            rejectTimeout(TxFinalizationTimeout),
+          ])
+        ).pipe(
+          mapRx((hash) => ({ hash, tx })),
+          retryWhen((errors) =>
+            errors.pipe(
+              tap((err) => console.error(` * Transaction failed: ${err}`)),
+              filterRx(
+                (err) => !err.toString().toLowerCase().includes("balance")
+              ),
+              delayWhen((_) => timer(RetryTimeout))
+            )
+          )
+        ),
+      ConcurrentTxLimit
+    )
+  );
+});
+
+/**
+ * Groups eras having some reward to be paid by validator stash accounts.
+ *
+ * @param {{ eras: *, pointsByEra: Object<string, *>, rewardsByEra: Object<string, *>, prefsByEra: Object<string, *> }} erasInfo
+ * @param {Array<BN>} validatorEras
+ * @returns {Object<string, { era: BN, eraReward: BN, reward: BN }>}
+ */
+const buildValidatorRewards = curry(
+  ({ pointsByEra, rewardsByEra, prefsByEra }, validatorEras) => {
+    const stashErasReducer = (acc, { eras, stashAddress }) => {
+      const eraReducer = (acc, era) => {
+        const eraKey = era.toString();
+        const eraPoints = pointsByEra[eraKey];
+        const eraRewards = rewardsByEra[eraKey];
+        const eraPrefs = prefsByEra[eraKey];
+
+        if (
+          eraPoints?.eraPoints.gt(new BN(0)) &&
+          eraPoints?.validators[stashAddress] &&
+          eraRewards &&
+          eraPrefs.validators[stashAddress]?.commission
+            ?.toBn()
+            .lt(MaxCommission)
+        ) {
+          const reward = eraPoints.validators[stashAddress]
+            .mul(eraRewards.eraReward)
+            .div(eraPoints.eraPoints);
+
+          if (!reward.isZero())
+            return acc.concat({
+              era,
+              eraReward: eraRewards.eraReward,
```

Comment:
Do you need `eraReward` and `reward` (apart from debugging) as you are not showing this data anyway?

Reply:
`reward` for each account will be shown in logs, see https://github.com/docknetwork/sdk/pull/286/files#diff-4cf85c6f9e9ab3caf77a6239964ea82cf0e8c404e7abe067e38bc8468f108dcbR167
`eraReward` was removed as it's currently unused.

Reply:
I had seen the logs but I was saying if you could remove the reward as its not needed. Anyway, its just a script so efficiency doesn't matter much so if it helps to keep it then lets keep it.